### PR TITLE
feat: add job for running service in e2e environment

### PR DESCRIPTION
**Pull request by effervescentia**

### CODE REVIEW

File: orb.yml [hunk 1]
```diff
@@ -3430,6 +3456,133 @@ jobs:
...
+              /bin/bash -c "yarn init:e2e"
+      - when:
+          condition: << parameters.e2e_service_dependencies >>
+          steps:
+            - run:
+                name: Start E2E dependencies
+                working_directory: ~/code/infrastructure-e2e
+                command: |
+                  docker wait cert-generator
+                  docker-compose -p vf -f docker-compose-vf.yaml up -d server-data-api-e2e
```
Comment (xavidop) [inline orb.yml:3543]: why are you removing server-data-api here? should those be the dependency services?

File: orb.yml [hunk 2]
```diff
@@ -3430,6 +3456,133 @@ jobs:
...
+              168387678261.dkr.ecr.us-east-1.amazonaws.com/ci-e2e-image:v1 \
+              /bin/bash -c "npx wait-on -t 60000 https://<< parameters.service_name >>.test.e2e:<< parameters.service_port >>/health"
+      - run:
+          name: Get logs
+          working_directory: ~/logs
+          when: on_fail
+          command: |
+            # Getting the logs
+            set +e
+            docker logs server-data-api-e2e > server-data-api.log
```
Comment (xavidop) [inline orb.yml:3579]: why it is server-data-api hard-coded here?

File: orb.yml [hunk 3]
```diff
@@ -2258,9 +2282,11 @@ commands:
           background: true
           command: |  
             # Set Node 16
-            nvm install v16.13.0
-            nvm use v16.13.0
-            nvm alias default v16.13.0
+            <<# parameters.use_nvm >>
```
Comment (xavidop) [inline orb.yml:2285]: why is this change needed? if you are on a docker executor you have to use the command setup_vf_dbs_docker. The setup_vf_dbs_machine is designed to use in machine executors.

**Review by xavidop — CHANGES_REQUESTED** [review]

File: orb.yml [hunk 4]
```diff
@@ -32,6 +32,26 @@ executors:
         - DEBUG=1
       - image: circleci/mongo:4.4.1 # MongoDB service container
       - image: redis:6.2-alpine # Redis service container
+  
+  e2e-check-executor:
```
[the remark this answers is not in the document]
Comment (effervescentia) [reply orb.yml:36]: I just need an executor that's similar to the one above but without all of the databases so that they can be run from the `infrastructure-e2e` repository instead

File: orb.yml (hunk 3 above)
Comment (effervescentia) [reply orb.yml:2285]: I need to skip using `nvm` because we already have the correct node version installed in the `168387678261.dkr.ecr.us-east-1.amazonaws.com/ci-e2e-image` image and the setup of `nvm` in the docker container was proving difficult

File: orb.yml (hunk 1 above)
Comment (effervescentia) [reply orb.yml:3543]: we're bringing up the `server-data-api-e2e` but yes it should be the `<< parameters.e2e_service_dependencies >>` here

File: orb.yml (hunk 2 above)
Comment (effervescentia) [reply orb.yml:3579]: I was actually going to hardcode all the services here so we can re-use this command across `test_e2e` and here, I'll do that now

File: orb.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (xavidop) [reply orb.yml:36]: in that case, you need to use the node-executor instead

File: orb.yml (hunk 2 above)
Comment (xavidop) [reply orb.yml:3579]: thank you!
